### DISCUSSION TITLE
Statistics on how long the battery lasted

### DISCUSSION THREAD

**Discussion by GegoSK** [andrew-codechimp/HA-Battery-Notes#1263] [Ideas]
Hello,
I briefly read all the discussions here, so I hope it is not mentioned somewhere. 

I think simple idea. When I press button "Battery replaced", the date in state is changed. Previous state will be forgoten. The previous date could be saved in attribution. I am interested in date of change and how many days the battery lasted for maybe 5 last changies. 

Thank you

**Comment by andrew-codechimp** [comment]
It has been mentioned a bit before in longer discussions but good to start a fresh topic dedicated to it to try and get some thoughts.
Statistics aren't my strong point so I'd have to look into whether I could do anything with the current battery replaced date or make a new sensor like battery replaced age that gets a days value stored from the last time.
A number of attributes doesn't seem the right way to approach it as you'd still have to manually crunch numbers to get averages and whatever arbitrary number I chose wouldn't be right for everyone.
Certainly on my list when I get time and if anyone does statistical magic and knows how this could be approached I'd welcome input.
